src/components/donation/Steps/Submit/estimateDonation.ts

```diff
@@ -113,7 +137,65 @@ export async function estimateDonation({
       content = { type: "evm", val: tx };
     }
     // ///////////// ESTIMATE TX ///////////////
-    return estimateTx(content, wallet);
+    const txEstimate = await estimateTx(content, wallet);
+    if (!txEstimate) return null;
+
+    // ///////////// Sucessful simulation ///////////////
+    const [tokenUSDValue, feeUSDValue] = await Promise.all([
+      _usdValue(token.coingecko_denom),
+      _usdValue(txEstimate.fee.coinGeckoId),
+    ]);
+
+    const tokenUSDAmountDec = new Decimal(tokenUSDValue).mul(token.amount);
+    const amount: EstimateItem = {
+      name: "Amount",
+      cryptoAmount: {
+        value: token.amount,
+        symbol: token.symbol,
+      },
+      fiatAmount: tokenUSDAmountDec.toNumber(),
+      prettyFiatAmount: `$${humanize(tokenUSDAmountDec, 4)}`,
+    };
+
+    const feeUSDValueAmount = new Decimal(feeUSDValue).mul(
+      txEstimate.fee.amount
+    );
+    const transactionFee: EstimateItem = {
+      name: "Transaction fee",
+      cryptoAmount: {
+        value: txEstimate.fee.amount.toString(),
+        symbol: txEstimate.fee.symbol,
+      },
+      fiatAmount: feeUSDValueAmount.toNumber(),
+      prettyFiatAmount: `$${humanize(feeUSDValueAmount, 4)}`,
+    };
+
+    const { isFiscalSponsored, endowType } = recipient;
+    const angelGivingFeeRatePCT =
+      endowType === "charity" ? (isFiscalSponsored ? 5.8 : 2.9) : 0;
```

Comment:
Just noticed that there's a bug here wherein if the endowment is not a Charity then the the estimate will be wrong (0% for fee). This is my fault for not explaining the fee structure better in the ticket. 

The base fee (contract-level) of 1.5% is always applied. The crypto fee of 1.4% is applied for non-fiat donations to all types. Only the FS Fee of 2.9% is relevant to Charity types only. Here's what the fee total logic would roughly look like building up from a base fee:

```javascript
let baseFee = 1.5;
if (!fiat) {
  baseFee += 1.4;
}
if (isFiscalSponsored) { 
  baseFee += 2.9;
}
```

Reply:
@SovereignAndrey, instead of `AngelGiving Fee` what would the fees be called for ASTs ?

Reply:
> @SovereignAndrey, instead of `AngelGiving Fee` what would the fees be called for ASTs ?

@ap-justin I'd say `AngelImpact Fee`.

> update: also for fiat donations where currency is `EUR` | `GBP` | `AUD`, is it needed to convert first to `USD` before applying `1.5%` base fee ?

Yes, good catch! CoinGecko quotes are all in USD I think, so it would reason we'd need to either:
1. Convert to their non-USD fiat value to USD OR 
2. Have an FX rate to turn the token prices into the rough EUR/GBP/etc equivalent 

I would say Option 2 is better UX since they can see everything in their native currency still.

Fiat donation: 100 EUR
FX Rate (EUR/USD): 0.98
Token price: 1.00 USD
Token price (~EUR): 0.98 EUR
Fees can show in EUR as they are % based

This can all come in a separate PR though once we add Stripe in. :)